docs/usage/SHOPPING_LIST_SYNC.md

```diff
@@ -73,28 +73,40 @@ Therefore, in order to resolve the actual ids of those references in the sync pr
 
 ````java
-// Build a ShoppingListQuery for fetching shopping lists from a source CTP project with all the needed references expanded for the sync
-final ShoppingListQuery shoppingListQueryWithReferenceExpanded = ShoppingListReferenceResolutionUtils.buildShoppingListQuery();
+// Build an ShoppingListQuery for fetching shopping lists from a source CTP project without any references expanded for the sync:
+final ShoppingListQuery shoppingListQueryWithReferences = ShoppingListReferenceResolutionUtils.buildShoppingListQuery();
 
 // Query all shopping lists (NOTE this is just for example, please adjust your logic)
 final List<ShoppingList> shoppingLists =
     CtpQueryUtils
-        .queryAll(sphereClient, shoppingListQueryWithReferenceExpanded, Function.identity())
+        .queryAll(sphereClient, shoppingListQueryWithReferences, Function.identity())
         .thenApply(fetchedResources -> fetchedResources
             .stream()
             .flatMap(List::stream)
             .collect(Collectors.toList()))
         .toCompletableFuture()
         .join();
+````
+
+In order to transform and map the shoppingList, 
+Initialize [`ShoppingListReferenceTransformService`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/shoppinglists/service/ShoppingListReferenceTransformService.java) with `sphereClient` and cache(You can use your own cache implementation and pass the map).
+For cache implementation, you can refer an example class in the library - which implements the cache using caffeine library with an LRU based cache eviction strategy[`InMemoryReferenceIdToKeyCache`](https://github.com/commercetools/commercetools-sync-java/tree/master/src/main/java/com/commercetools/sync/commons/utils/InMemoryReferenceIdToKeyCache.java).
+Then call the transformShoppingListReferences method with the `shoppingLists` parameter as shown below:
 
-// Mapping from ShoppingList to ShoppingListDraft with considering reference resolution.
-final List<ShoppingListDraft> shoppingListDrafts = ShoppingListReferenceResolutionUtils.mapToShoppingListDrafts(shoppingLists);
+````java
+// Fetch(Id to key values for references) into the cache and map from ShoppingList to ShoppingListDraft using cache with considering reference resolution.
+final List<ShoppingListDraft> shoppingListDrafts = ShoppingListReferenceTransformService.transformShoppingListReferences(shoppingLists);
 ````
 
+The cache here is used for a better performance. 
+Instead of expanding the references in the query for shoppingList resource. `ShoppingListReferenceTransformService` will execute a query to fetch key values for the Ids(nonCached) and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```

Comment:
```suggestion
Instead of expanding the references in the query for shoppingList resource. `ShoppingListReferenceTransformService` will execute a query to fetch key-id pairs and store in cache. These cached id to key values then can be used by another resource for resolving its references instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve the overall performance of the sync library and commercetools API.
```